File: VB/AddRowsRuntime/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraVerticalGrid.Rows

Namespace AddRowsRuntime

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            CreateRows()
            Dim person As Person = New Person()
            person.LastName = "Doe"
            person.FirstName = "John"
            propertyGridControl1.SelectedObject = person
        End Sub

        Private Sub CreateRows()
            Dim rowLastName As EditorRow = New EditorRow("LastName")
            rowLastName.Properties.Caption = "Last name"
            propertyGridControl1.Rows.Add(rowLastName)
            Dim rowCategory As CategoryRow = New CategoryRow("Address")
            propertyGridControl1.Rows.Add(rowCategory)
            Dim rowAddressLine1 As EditorRow = New EditorRow("AddressLine1")
            rowAddressLine1.Properties.Caption = "Address Line 1"
            rowCategory.ChildRows.Add(rowAddressLine1)
            Dim rowZip As EditorRow = New EditorRow("Zip")
            rowZip.Properties.Caption = "Zip Code"
            rowCategory.ChildRows.Add(rowZip)
        End Sub
    End Class

    Public Class Person

        Private _FirstName As String

        Public Property FirstName As String
            Get
                Return _FirstName
            End Get

            Set(ByVal value As String)
                _FirstName = value
            End Set
        End Property

        Private _LastName As String

        Public Property LastName As String
            Get
                Return _LastName
            End Get

            Set(ByVal value As String)
                _LastName = value
            End Set
        End Property

        Private _AddressLine1 As String

        Public Property AddressLine1 As String
            Get
                Return _AddressLine1
            End Get

            Set(ByVal value As String)
                _AddressLine1 = value
            End Set
        End Property

        Private _Zip As String

        Public Property Zip As String
            Get
                Return _Zip
            End Get

            Set(ByVal value As String)
                _Zip = value
            End Set
        End Property
    End Class
End Namespace

File: CS/AddRowsRuntime/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraVerticalGrid.Rows;

namespace AddRowsRuntime {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            propertyGridControl1.AutoGenerateRows = false;
            CreateRows();
            Person person = new Person();
            person.LastName = "Doe";
            person.FirstName = "John";
            propertyGridControl1.SelectedObject = person;
        }

        private void CreateRows() {
            EditorRow rowLastName = new EditorRow("LastName");
            rowLastName.Properties.Caption = "Last name";
            propertyGridControl1.Rows.Add(rowLastName);
            CategoryRow rowCategory = new CategoryRow("Address");
            propertyGridControl1.Rows.Add(rowCategory);
            EditorRow rowAddressLine1 = new EditorRow("AddressLine1");
            rowAddressLine1.Properties.Caption = "Address Line 1";
            rowCategory.ChildRows.Add(rowAddressLine1);
            EditorRow rowZip = new EditorRow("Zip");
            rowZip.Properties.Caption = "Zip Code";
            rowCategory.ChildRows.Add(rowZip);
        }
    }

    public class Person {
        private string _FirstName;
        public string FirstName {
            get { return _FirstName; }
            set { _FirstName = value; }
        }

        private string _LastName;
        public string LastName {
            get { return _LastName; }
            set { _LastName = value; }
        }

        private string _AddressLine1;
        public string AddressLine1 {
            get { return _AddressLine1; }
            set { _AddressLine1 = value; }
        }

        private string _Zip;
        public string Zip {
            get { return _Zip; }
            set { _Zip = value; }
        }
    }
}
